### PR TITLE
Describe extended versioning and branches

### PR DESCRIPTION
**Pull request by vshlapakov**
Review, please.

---

This change is [Reviewable](https://reviewable.io/reviews/scrapinghub/scrapinghub-stack-scrapy/13)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -4,19 +4,67 @@ scrapinghub-stack-scrapy
...
+- latest version of the stack is matched with a branch name without ``branch`` prefix with ``-latest`` suffix
+
+  Examples:
+
+    - ``branch-1.0`` -> ``1.0-latest``
+    - ``branch-1.1`` -> ``1.1-latest``
+    - ``branch-1.1-py3`` -> ``1.1-py3-latest``
+    - ``branch-1.0-dev`` -> ``1.0-dev-latest``
+
+  Warning: please do not use stack marked with ``latest`` tag (i.e: ``scrapinghub-stack-scrapy:latest``), as it can contain an arbitrary stack version (latest deployed one), it's a part of Docker build system and we can't disable it for now.
```
Comment (chekunkov) [inline README.rst]: we should discourage use of any ``latest`` stacks (``latest``, ``1.1-latest``, ``1.1-py3-latest`` etc)

File: README.rst [hunk 2]
```diff
@@ -32,20 +80,23 @@ When you're going to release a new version of the stack, you should:
...
 
 3. Push the changes and the tags to the repo::
 
-    git push -f origin master <tag1> <tag2>
+    git push -f origin master <tag1> <tag2> <tag3>
 
   Example::
 
-    git push -f origin master 1.1 1.1-20160731
+    git push -f origin master 1.1-py3 1.1-py3-20160804 1.1-py3-latest
```
Comment (chekunkov) [inline README.rst]: 1.1-py3-latest is built automatically from branch-1.1-py3, no need to define a tag and push it

File: README.rst [hunk 3]
```diff
@@ -4,19 +4,67 @@ scrapinghub-stack-scrapy
...
+
+  Example:
+
+    - ``1.0``
+    - ``1.1``
+    - ``1.1-py3``
+
+- each published version of the stack is marked with ``<scrapy version>[-py3]-<release date>`` tag
+
+  It's highly recommended to use these versions (or major ones above) over others.
```
Comment (chekunkov) [inline README.rst]: Should be really visible, let's use bold text?

File: README.rst [hunk 4]
```diff
@@ -32,20 +79,24 @@ When you're going to release a new version of the stack, you should:
...
-    git tag 1.1-20160731
+    git tag -f 1.1-py3
+    git tag 1.1-py3-20160804
+
+  Tag ``1.1-py3-latest`` will be built automatically from ``branch-1.1-py3``, no need to define a tag manually.
 
 3. Push the changes and the tags to the repo::
 
-    git push -f origin master <tag1> <tag2>
+    git push -f origin master <tag1> <tag2> <tag3>
```
Comment (chekunkov) [inline README.rst]: `git push -f origin <branch> <tag1> <tag2> <tag3>`

I'm not sure what kind of development will go in `master` - it seems to me that we will do development in parallel in 3 major branches.

File: README.rst [hunk 5]
```diff
@@ -32,20 +79,24 @@ When you're going to release a new version of the stack, you should:
...
 
   Example::
 
-    git push -f origin master 1.1 1.1-20160731
+    git push -f origin master 1.1-py3 1.1-py3-20160804
```
Comment (chekunkov) [inline README.rst]: if you push 1.1-py3 tags - I expect example to show push to `branch-1.1-py3` branch

Comment (vshlapakov) [inline README.rst]: A good point, updated, thanks!

**Comment by chekunkov** [comment]
@vshlapakov looks good! now let's merge this to all major branches ;)

**Pull request merged**